genie-core/src/main/java/com/netflix/genie/core/jobs/JobConstants.java

```diff
@@ -269,12 +280,14 @@
         .append("trap \"handle_failure\" ERR EXIT\n\n")
         .append("function handle_failure {\n")
         .append("    ERROR_CODE=$?\n")
+        .append("    echo \"Handling exit signal (code: ${ERROR_CODE})\"\n")
+        .append("\n")
         .append("    # Good exit\n")
         .append("    if [[ ${ERROR_CODE} -eq 0 ]]; then\n")
         .append("        exit 0\n")
         .append("    fi\n")
         .append("    # Bad exit\n")
-        .append("    printf '{\"exitCode\": \"%s\"}\\n' \"${ERROR_CODE}\" > ./genie/genie.done\n")
+        .append("    printf '{\"exitCode\": \"%s\"}\\n' \"${ERROR_CODE}\" > ${GENIE_JOB_DIR}/genie/genie.done\n")
```

Comment:
Since we use `/genie` a lot perhaps we should make it its own variable. Like `${GENIE_LOG_DIR}` or something? Just a convenience thought.

Also I think there is a String variable for the name of the done file and we probably should use that here instead of the genie.done string. That way in case we change it it changes here too.